src/app/components/admin/admin-evaluations/admin-evaluations.component.ts

```diff
@@ -196,12 +217,59 @@ export class AdminEvaluationsComponent implements OnInit, OnDestroy {
     });
   }
 
-  applyFilter(filterValue: string) {
-    this.filterControl.setValue(filterValue);
+  applyFilter() {
+    this.filteredEvaluationList = this.evaluationList.filter(evaluation =>
+      !this.filterString ||
+      evaluation.description.toLowerCase().includes(this.filterString)
+    );
+
+    this.sortChanged(this.sort);
+  }
+
+  clearFilter() {
+    this.filterString = '';
+    this.filterControl.setValue('');
+    this.loadInitialData();

```

Comment:
Calling loadInitialData() here adds another listener for evaluations.  The reason this works to refresh the evaluation list is that the listener fires to get the current value when it is added.  However, each time clearFilter() is run, it will add another listener, which means the code in lines 107-108 gets run multiple times (number of clearFilter clicks + 1).  Instead of calling loadInitialData(), this should call applyFilter(), which resets the filteredEvaluationList, without adding another listener.